src/status_im/extensions/capacities/ethereum.cljs

```diff
@@ -42,35 +37,11 @@
                       #(f db (assoc arguments address-keyword %))))
       (f db arguments))))
 
-(defn prepare-extension-transaction [params contacts on-success on-failure]
-  (let [{:keys [to value data gas gasPrice nonce]} params
-        contact (get contacts (hex/normalize-hex to))]
-    (cond-> {:id               "extension-id"
-             :to-name          (or (when (nil? to)
-                                     (i18n/label :t/new-contract))
-                                   contact)
-             :symbol           :ETH
-             :method           constants/web3-send-transaction
-             :to               to
-             :amount           (money/bignumber (or value 0))
-             :gas              (cond
-                                 gas
-                                 (money/bignumber gas)
-                                 (and value (empty? data))
-                                 (money/bignumber 21000))
-             :gas-price        (when gasPrice
-                                 (money/bignumber gasPrice))
-             :data             data
-             :on-result        [:extensions/wallet-ui-on-success on-success]
-             :on-error         [:extensions/wallet-ui-on-failure on-failure]}
-      nonce
-      (assoc :nonce nonce))))
-
 (defn- execute-send-transaction [db {:keys [method params on-success on-failure] :as arguments}]
-  (let [tx-object (assoc (select-keys arguments [:to :gas :gas-price :value :nonce])
-                         :data (when (and method params) (abi-spec/encode method params)))
-        transaction (prepare-extension-transaction tx-object (:contacts/contacts db) on-success on-failure)]
-    (wallet/open-modal-wallet-for-transaction db transaction tx-object)))
+  (signing/sign {:db db} {:tx-obj    (assoc (select-keys arguments [:to :gas :gas-price :value :nonce])
```

Comment:
why not just transaction? so far we have been pretty strict about not using abbreviations in the code base and I don't see how it is an object

Reply:
it's object from here https://github.com/ethereum/wiki/wiki/JavaScript-API#parameters-25, Object - The transaction object to send:
but i'm open for suggestions on better naming